spec/controllers/admin/contest_groups_controller_spec.rb

```diff
@@ -0,0 +1,179 @@
+RSpec.describe Admin::ContestGroupsController, type: :controller do
+  context 'without a logged in user' do
+    describe 'GET #index' do
+      before { get :index }
+
+      it { should redirect_to(new_session_path) }
+    end
+
+    # TODO add tests for the other actions
+  end
+
+  shared_examples "accessed by an unauthorized user" do
+    # TODO add tests for coach and contester
+  end
+
+  context 'with an admin user' do
+
+    let(:user) { create(:user, role: User::ADMIN) }
+
+    before { sign_in(user) }
+
+    let(:first_contest_group) { create :contest_group }
+    let(:second_contest_group) { create :contest_group }
+
+    context 'routing' do
+      it { should route(:get,    '/admin/contest_groups').to        action: :index }
+      it { should route(:get,    '/admin/contest_groups/new').to    action: :new }
+      it { should route(:post,   '/admin/contest_groups').to        action: :create }
+      it { should route(:get,    '/admin/contest_groups/1/edit').to action: :edit, id: 1 }
+      it { should route(:put,    '/admin/contest_groups/1').to      action: :update, id: 1}
+      it { should route(:delete, '/admin/contest_groups/1').to      action: :destroy, id: 1 }
+    end
+
+    it { should use_before_action(:set_contest_group) }
```

Comment:
I don't like this kind of validations in the specs, as IMO they couple the implementation with the specs too much.